From 45c65ebeefcb20c2d2480b11b299d49f288c3b73 Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul.cercueil@analog.com>
Date: Mon, 12 Oct 2015 12:19:42 +0200
Subject: [PATCH 05/22] gr-qtgui: Add a workaround for an upstream bug of 'uic'

The Qt5 version of the 'uic' compiler generates incorrect code right now.
The bug has been reported (https://bugreports.qt.io/browse/QTBUG-48492) and
is pending review.

Signed-off-by: Paul Cercueil <paul.cercueil@analog.com>
---
 gr-qtgui/include/QtWidgets/QRegExpValidator | 9 +++++++++
 1 file changed, 9 insertions(+)
 create mode 100644 gr-qtgui/include/QtWidgets/QRegExpValidator

diff --git a/gr-qtgui/include/QtWidgets/QRegExpValidator b/gr-qtgui/include/QtWidgets/QRegExpValidator
new file mode 100644
index 0000000000..17f6ce8333
--- /dev/null
+++ b/gr-qtgui/include/QtWidgets/QRegExpValidator
@@ -0,0 +1,9 @@
+/*
+ * The Qt5 version of the 'uic' compiler generates incorrect code right now.
+ * The bug has been reported (https://bugreports.qt.io/browse/QTBUG-48492) and
+ * is pending review.
+ *
+ * This file will make sure that the build of gr-qtgui will succeed with the
+ * broken version of 'uic'.
+ */
+#include <QtGui/QRegExpValidator>
-- 
2.11.0

